src/goto-instrument/cover_filter.cpp

```diff
@@ -33,10 +33,7 @@ bool internal_functions_filtert::operator()(
     return false;
 
   // ignore Java built-ins
-  if(
-    has_prefix(id2string(identifier), "java::array[") ||
-    has_prefix(id2string(identifier), "java::org.cprover") ||
-    has_prefix(id2string(identifier), "java::java."))
+  if(has_prefix(id2string(identifier), "java::array["))
```

Comment:
The comment above seemed to fit the previous code, but now this is really unclear. What's so special about `array[`?